sdk/lib/core/list.dart

```diff
@@ -537,6 +537,8 @@ abstract class List<E> implements EfficientLengthIterable<E> {
   /**
    * Pops and returns the last object in this list.
    *
+   * An error occurs if the list is empty.
```

Comment:
Just say: 
```
 * The list must not be empty.
```

This is how we document prerequisites in the platform libraries. 
If a "must" requirement is not satisfied, the implementation will *likely* throw an `Error`, but there are situations where the throwing is best-effort only (mainly concurrent modification).